src/drivers/github_test.go

```diff
@@ -4,15 +4,226 @@ import (
 	"encoding/json"
 	"io/ioutil"
 	"net/http"
+	"testing"
 
 	. "github.com/Originate/git-town/src/drivers"
+	"github.com/stretchr/testify/assert"
 	httpmock "gopkg.in/jarcoal/httpmock.v1"
-
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
 )
 
-func GetRequestData(request *http.Request) map[string]interface{} {
+var pullRequestBaseURL = "https://api.github.com/repos/Originate/git-town/pulls"
+var currentPullRequestURL = pullRequestBaseURL + "?base=main&head=Originate%3Afeature&state=open"
+var childPullRequestsURL = pullRequestBaseURL + "?base=feature&state=open"
+var mergePullRequestURL = pullRequestBaseURL + "/1/merge"
+var updatePullRequestBaseURL1 = pullRequestBaseURL + "/2"
+var updatePullRequestBaseURL2 = pullRequestBaseURL + "/3"
+
+func setupDriver(t *testing.T, token string) (CodeHostingDriver, func()) {
+	httpmock.Activate()
+	driver := GetDriver(DriverOptions{OriginURL: "git@github.com:Originate/git-town.git"})
+	assert.NotNil(t, driver)
+	if token != "" {
+		driver.SetAPIToken(token)
+	}
+	return driver, func() {
+		httpmock.DeactivateAndReset()
+	}
+}
+
+func TestGitHubDriver_CanMergePullRequest(t *testing.T) {
+	driver, teardown := setupDriver(t, "TOKEN")
+	defer teardown()
```

Comment:
To address one of your concerns, I have dried up the repetitive test setup logic by extracting the setup of the Github driver into a `setupDriver` function and a `teardown` function here. This test is the only one which I think needs this. The other tests are simple enough to be simple functions. 

I like this setup better than the [Ginkgo version](https://github.com/Originate/git-town/blob/master/src/drivers/github_test.go) because:
- the Ginkgo version had multiple layers of nesting of `Describe` blocks
- some tests had multiple nested `BeforeEach` blocks that apply to them
- variables were shared across `BeforeEach` blocks and several test blocks, which are sometimes pages apart from each other. One had to rely on test framework magic that these variables aren't shared between test runs.
- it is now explicit and debuggable when which setup code is called in each test script

Reply:
Also, instead of naming this `beforeEach`, the shared test setup code has a name that describes what it does: `setupDriver`.